### ISSUE TITLE
Right click does 'Copy & Paste' in MacOS with 'Right click' set to 'Paste' in 'Mouse' settings

### ISSUE THREAD

**Issue by junwu-cao** [Eugeny/tabby#9237]
Version: 1.0.204
Platform: macOS x64 14.1.1
Plugins: none
Frontend: xterm-webgl

I'm running tabby 1.0.204 on MacBook Pro (MacOS 14.1.1) and the 'Right click' in 'Mouse' settings is set to 'Paste'. When I mark a text string and right-click mouse button without moving mouse pointer, the text will be pasted at the cursor position as expected. If I moved the mouse pointer to another text string, then right-click will actually copy the text where mouse pointer points to and paste it at the cursor position instead of the text marked previously.

This behavior doesn't happen on Windows 10/11 (same Tabby version) and I think it is not correct since 'Right click' of 'Mouse' is set to 'Paste', not anything else. By the way, this also happens in previous Tabby versions for MacOS.

Thanks.

![tabby 00](https://github.com/Eugeny/tabby/assets/126440335/a5773a87-1f2d-4e54-a451-8a287b888afe)
![tabby 01](https://github.com/Eugeny/tabby/assets/126440335/4bedf69d-e8aa-42b6-93c1-84fc7a480f7e)
![tabby 02](https://github.com/Eugeny/tabby/assets/126440335/43386be5-5c1d-4347-9e7d-9a2cb89d34a9)

**Comment by junwu-cao** [comment]
Yes "copy on select" option is enabled. This is true for both of my Windows and MacOS. The thing is the right-click should not select any text where the mouse pointer points to. It should just paste whatever I selected before I moved my mouse pointer. This is the case for Windows 10/11 (as expected).

But right-click on MacOS always select the text where the mouse pointer points to no matter what (unless I move my mouse pointer to an empty space). So, the right-click actually performs two actions, select the text and paste it. 'Right click' in 'Mouse' settings is set to 'Paste', nothing else. This only happens in my MacOS.

![Tabby_Settings_Win](https://github.com/Eugeny/tabby/assets/126440335/b8337ba6-0449-4913-9a20-856808d6f058)
![Tabby_Settings_MacOS](https://github.com/Eugeny/tabby/assets/126440335/105cc271-b71e-4cd2-b695-fbfdd1764919)

**Comment by Eugeny** [comment]
Selecting the word when right-clicking it is default macOS behaviour, but I see why one might want to disable this. I'll convert this ticket into an improvement.

**Comment by junwu-cao** [comment]
Thank you so much.
